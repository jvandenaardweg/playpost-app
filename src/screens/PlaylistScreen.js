import React from 'react';
import { PlaylistsContainer } from '@/containers/PlaylistsContainer';

import { AppBackground } from '@/components/AppBackground';

export class PlaylistScreen extends React.Component {
  static navigationOptions = {
    title: 'Playlist',
  };

  render() {
    return (
      <AppBackground>
        <PlaylistsContainer />
      </AppBackground>
    );
  }
}
