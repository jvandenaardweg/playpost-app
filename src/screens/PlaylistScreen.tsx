import React from 'react';
import { NavigationScreenProp, NavigationRoute } from 'react-navigation';

import { PlaylistsContainer } from '../components/Playlists';
import { AppBackground } from '../components/AppBackground';
import { ButtonUpgrade } from '../components/Header/ButtonUpgrade';

interface Props {
  navigation: NavigationScreenProp<NavigationRoute>;
}
export class PlaylistScreen extends React.PureComponent<Props> {
  static navigationOptions = ({ navigation }: { navigation: NavigationScreenProp<NavigationRoute> }) => {
    return {
      title: 'Playlist',
      headerRight: <ButtonUpgrade onPress={navigation.getParam('handleOnPressUpgrade')} />
    };
  }

  componentDidMount() {
    this.props.navigation.setParams({ handleOnPressUpgrade: this.handleOnPressUpgrade });
  }

  handleOnPressUpgrade = () => {
    this.props.navigation.navigate('Upgrade');
  }

  render() {
    return (
      <AppBackground>
        <PlaylistsContainer />
      </AppBackground>
    );
  }
}
