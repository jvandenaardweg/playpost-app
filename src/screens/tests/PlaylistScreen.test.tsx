import React from 'react';
// tslint:disable-next-line: no-submodule-imports
import * as ShallowRenderer from 'react-test-renderer/shallow';

import { PlaylistScreen } from '../PlaylistScreen';

describe('PlaylistScreen', () => {

  describe('rendering', () => {
    let wrapper: any;

    beforeEach(() => {
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(<PlaylistScreen />);
      wrapper = renderer.getRenderOutput();
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
