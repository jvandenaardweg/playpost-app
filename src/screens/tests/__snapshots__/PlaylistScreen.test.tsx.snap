// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistScreen rendering should render correctly 1`] = `
<Memo>
  <InteractionManaged
    isAnimated={true}
    showActivityIndicator={true}
  >
    <Memo(Connect(PlaylistContainerComponent)) />
  </InteractionManaged>
</Memo>
`;
